firewood/src/manager.rs

```diff
@@ -123,6 +122,13 @@ impl RevisionManager {
         // from opening the same database simultaneously
         fb.lock()?;
 
+        let root_store: Box<dyn RootStore + Send + Sync> = match config.root_store_dir {
+            Some(path) => {
+                Box::new(FjallStore::new(path).map_err(RevisionManagerError::RootStoreError)?)
+            }
+            None => Box::new(NoOpStore {}),
```

Comment:
Its starting to feel like we don't need `NoOpStore` and could instead implement `RootStore` on `Option<T> where T: RootStore` that does the same thing as `NoOpStore` if the option is `None` (which may also require a pass-through implementation of `RootStore` on `Box<T>` OR `Box<dyn RootStore + Send + Sync>` (but not both).